packages/workbox-core/src/types.ts

```diff
@@ -72,12 +72,23 @@ export interface RouteHandlerObject {
  */
 export type RouteHandler = RouteHandlerCallback | RouteHandlerObject;
 
+export interface HandlerWillStartCallbackParam {
+  request: Request;
+  event?: ExtendableEvent;
+  state?: {[prop: string]: any};
```

Comment:
How about creating a `State` interface with that signature, and then using it throughout instead of repeating ` {[prop: string]: any}` multiple times?

That could make it easier to change things if we ever decide to get stricter about the shape of `State`.

Reply:
This is really just an object that could accept any arbitrary keys and values, but TypeScript doesn't support that easy (it tries to force you to be specific).

I'm not opposed to making an interface that makes that explicit though (which we could reuse).

Reply:
I'll add a generic `MapLike` object with this type, since we also use it for `params`.

Reply:
Sure. Added a `PluginState` type just for those cases.